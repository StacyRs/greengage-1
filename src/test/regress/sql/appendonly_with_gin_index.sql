-- Given I have an append-only table
create table users(
  first_name tsvector
) with (appendonly=true);

-- And I have a large amount of data in the table
insert into users
  select to_tsvector( md5(random()::text))
  from generate_series(1, 60000) i;

insert into users values (to_tsvector('John'));

-- When I create a GIN index on users
CREATE INDEX users_search_idx ON users USING gin (first_name);

-- Orca performs seq scan in this case, so disable Orca.
set optimizer = 0;

-- Then I should be able to query the table
select * from users where first_name @@ to_tsquery('John');
explain (costs off) select * from users where first_name @@ to_tsquery('John');

drop table users;
reset optimizer;
