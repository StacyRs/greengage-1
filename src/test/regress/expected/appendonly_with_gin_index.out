-- Given I have an append-only table
create table users(
  first_name tsvector
) with (appendonly=true);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause, and no column type is suitable for a distribution key. Creating a NULL policy entry.
-- And I have a large amount of data in the table
insert into users
  select to_tsvector( md5(random()::text))
  from generate_series(1, 60000) i;
insert into users values (to_tsvector('John'));
-- When I create a GIN index on users
CREATE INDEX users_search_idx ON users USING gin (first_name);
-- Orca performs seq scan in this case, so disable Orca.
set optimizer = 0;
-- Then I should be able to query the table
select * from users where first_name @@ to_tsquery('John');
 first_name 
------------
 'john':1
(1 row)

explain (costs off) select * from users where first_name @@ to_tsquery('John');
                          QUERY PLAN                           
---------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   ->  Bitmap Heap Scan on users
         Recheck Cond: (first_name @@ '''john'''::tsquery)
         ->  Bitmap Index Scan on users_search_idx
               Index Cond: (first_name @@ '''john'''::tsquery)
 Optimizer: Postgres query optimizer
(6 rows)

drop table users;
reset optimizer;
